photopicker/photopicker.js

```diff
@@ -0,0 +1,138 @@
+'use strict';
+
+if (!flickrAPI) {
+  var flickrAPI = {};
+}
+
+if (!Promise) {
+  var Promise = {};
+}
+
+var URL_STEM = 'https://api.flickr.com/services/rest/?';
+
+/**
+ * Pulls a list of the Top 100 Countries with Photos from Flickr API.
+ */
+function getTopCountriesList() {
+  var request = $.ajax({
+    data: {
+      method: 'flickr.places.getTopPlacesList',
+      api_key: flickrAPI,
+      place_type_id: 12,
+      format: 'json',
+      nojsoncallback: 1
+    },
+    dataType: 'json',
+    url: URL_STEM,
+    async: false
+  });
+  return JSON.parse(request.responseText);
+}
+
+/**
+ * Pulls a list of places with public photos that are geotagged for the randomly
+ * chosen country from the FlickrAPI. Returns an array that contains the list
+ * as object as well as the name of the country.
+ */
+function getPlacesByCountry(topPlacesList) {
+  var countryID = _.sample(topPlacesList.places.place);
+  var country = countryID.woe_name;
+  var request = $.ajax({
+    data: {
+      method: 'flickr.places.getChildrenWithPhotosPublic',
+      api_key: flickrAPI,
+      place_id: countryID.place_id,
+      format: 'json',
+      nojsoncallback: 1
+    },
+    dataType: 'json',
+    url: URL_STEM,
+    async: false
+  });
+  var placesByCountry = JSON.parse(request.responseText);
+  return [placesByCountry, country];
+}
+
+/**
+ * Pulls all photos geotagged in the randomly chosen place from the FlickrAPI.
+ * Returns and array that contains the photo list as well as the longitude and
+ * latitude of the place.
+ */
+function getPhotosByPlace(placesByCountry) {
+  var placeID = _.sample(placesByCountry.places.place);
+  var request = $.ajax({
+    data: {
+      method: 'flickr.photos.search',
+      api_key: flickrAPI,
+      place_id: placeID.place_id,
+      format: 'json',
+      nojsoncallback: 1
+    },
+    dataType: 'json',
+    url: URL_STEM,
+    async: false
+  });
+  return JSON.parse(request.responseText);
+}
+
+/**
+ * Gets the coordinates for the given photo from flickr API
+ */
+function getPhotoLocation(photoid) {
```

Comment:
That's how those particular argument are called as in the API parameters. Wouldn't recognize them if they were in camelCase